### DISCUSSION TITLE
Great template, but limitations with SSR and dynamic content handling

### DISCUSSION THREAD

**Discussion by AndresBarretoDev** [arthelokyo/astrowind#346] [General]
Hello everyone,

I wanted to express my admiration for this template; it has many great features. However, I've encountered a significant issue: it seems not to work for SSR (Server-Side Rendering). I tried handling dynamic content from a CMS, but unfortunately, this proved to be impossible.

Despite attempting to use it in Hybrid or Server mode, I consistently ran into errors. During the build process, it appears to add code from the config.ts file, resulting in it not functioning properly in production.

In summary, the template is excellent for static content, but for those of us who need to manage dynamic content and utilize SSR, it may not be the best choice. I would appreciate any guidance or solutions you can provide to overcome this obstacle.

Thank you!

![image](https://github.com/onwidget/astrowind/assets/25916732/a60d754e-a5b1-445e-8b29-8fd0ef17b943)

This is an example of compilation. However, when compiling in this way and deploying it to Vercel, it throws errors such as:

![image](https://github.com/onwidget/astrowind/assets/25916732/6bcd0b31-f4a2-4c13-9d9e-a899e04368a3)

And this error is due to the fact that, understandably, during the compilation, there is a file with this embedded code.

![image](https://github.com/onwidget/astrowind/assets/25916732/98f99c02-e89e-4491-bb96-0f530abab3a5)

**Comment by SamuelAierizer** [comment]
Encountered the same problem. I didn't even think this template may not support SSR. Spent days building my site and then whops, SSR is no supported and no easy way to patch things up. 

Should be written more clearly somewhere.

**Comment by prototypa** [comment]
Hi @AndresBarretoDev and @SamuelAierizer 

It is true that this template does not support SSR and was not explicitly declared in the README. We have already added it:

https://github.com/onwidget/astrowind/blob/main/README.md#getting-started

@AndresBarretoDev, we have connected the template **AstroWind** to different CMS in several projects but by rewriting the `utils/blogs.js` file

https://github.com/onwidget/astrowind/blob/main/src/utils/blog.ts

We're organizing that code a bit to include in future versions of **AstroWind**, as well as SSR support.

**Comment by SamuelAierizer** [comment]
For those who need a quick fix for this, i was able to get around it by converting my .yaml file to a json and have something like this: 

```
const config = {
  'name' : ....
 }
```

This kind of defeats the purpose but from what I can tell until now, it was a somewhat working patch. With this change I was able to deploy to vercel and use SSR.

From what I understood it is a very bad practice to open files like this when you have SSR and a better workaround would be to make the config file available via a GET route and then make a request to the config file and process it from there. I don't know how this would impact the theme as a whole, maybe there is a better way to solve this. 

Don't get me wrong I love your work and I think it's one of the best out there. Was just surprised when I encountered this issue.

**Comment by prototypa** [comment]
Hi @SamuelAierizer 

I agree that the `config yaml` file does not work for SSR. SSR was not in the initial objective of the template, in fact, at the time we created the template not even Astro had support for SSR, for that reason many things have a different engineering and it takes developing a new version designed for the version of current Astro that we know. Many of the things about **AstroWind** in its time were developed to supply features that Astro did not even have. Some are unnecessary at this point, however for static sites it works. Thanks for the feedback, it is something we plan to improve very soon.

**Comment by prototypa** [comment]
Hi @AndresBarretoDev and @SamuelAierizer 

We have made some changes to process `config.yaml` at build time. For this and other needs we have developed an Astro integration within `src/integration` (for now local, in version 2.0 it will be moved to another package) that takes care of this and exposes a virtual import of vite `"astrowind:config"`.

If you can take a look and see if that works for you.

PS: The integration will also help us in the **AstroWind 2.0** version to encapsulate all the useful components and functions that support the template and do not depend on the particular needs, content and UI of use by each user. We have always had a problem with how to update the template. One path will be that, and AstroWind integration + a cli command to generate Widgets and Components or update them.
